### ISSUE TITLE
Bug in spack checksum?

### ISSUE THREAD

**Issue by adamjstewart** [LLNL/spack#2947]
```
$ spack checksum r-rcpp
==> Found 73 versions of r-rcpp:
  
  0.14.4  https://cran.r-project.org/src/contrib/cladoRcpp_0.14.4.tar.gz
  0.12.9  https://cran.r-project.org/src/contrib/Rcpp_0.12.9.tar.gz
  0.12.8  https://cran.r-project.org/src/contrib/Archive/Rcpp/Rcpp_0.12.8.tar.gz
  0.12.7  https://cran.r-project.org/src/contrib/Archive/Rcpp/Rcpp_0.12.7.tar.gz
...
```
`cladoRcpp` shouldn't be picked up.

**Comment by tgamblin** [comment]
You want to look at [`Version.wildcard()`](https://github.com/LLNL/spack/blob/develop/lib/spack/spack/version.py#L198).  That's how we currently generate a regex to search for additional download URLs.  It's not perfect, and all this scraping stuff is more art than science.

I think one approach you could take to eliminate false positives like this would be to rip `wildcard()` out of `Version` and put it in `url`, then generate the wildcards based on the more specialized regexes that match the original URL.

**Comment by adamjstewart** [comment]
Similar case:
```
$ spack create https://cran.r-project.org/src/contrib/mda_0.4-9.tar.gz
==> This looks like a URL for mda
==> Found 2 versions of mda:
  
  1.1.4  https://cran.r-project.org/src/contrib/survJamda_1.1.4.tar.gz
  0.4-9  https://cran.r-project.org/src/contrib/mda_0.4-9.tar.gz
```

**Comment by adamjstewart** [comment]
A particularly bad example:
```
$ spack create https://cran.r-project.org/src/contrib/pls_2.6-0.tar.gz
==> This looks like a URL for pls
==> Found 8 versions of pls:
  
  5.7    https://cran.r-project.org/src/contrib/enpls_5.7.tar.gz
  2.6-0  https://cran.r-project.org/src/contrib/pls_2.6-0.tar.gz
  2.2-1  https://cran.r-project.org/src/contrib/spls_2.2-1.tar.gz
  1.6-1  https://cran.r-project.org/src/contrib/ppls_1.6-1.tar.gz
  1.4    https://cran.r-project.org/src/contrib/svapls_1.4.tar.gz
  1.3    https://cran.r-project.org/src/contrib/autopls_1.3.tar.gz
  1.0.4  https://cran.r-project.org/src/contrib/matrixpls_1.0.4.tar.gz
  0.2-1  https://cran.r-project.org/src/contrib/lspls_0.2-1.tar.gz
```
From what I gather, our regex is looking for any file named `*name_version.tar.gz`. Removing the `*` from the front should help.

**Comment by adamjstewart** [comment]
This also affected some X.org packages where all of the tarballs are in a single directory:
```
$ spack checksum xproto
==> Found 32 versions of xproto:
  
  7.0.31    https://www.x.org/archive/individual/proto/xproto-7.0.31.tar.gz
  7.0.30    https://www.x.org/archive/individual/proto/xproto-7.0.30.tar.gz
  7.0.29    https://www.x.org/archive/individual/proto/xproto-7.0.29.tar.gz
  7.0.28    https://www.x.org/archive/individual/proto/xproto-7.0.28.tar.gz
  7.0.27    https://www.x.org/archive/individual/proto/xproto-7.0.27.tar.gz
  7.0.26    https://www.x.org/archive/individual/proto/xproto-7.0.26.tar.gz
  7.0.25    https://www.x.org/archive/individual/proto/xproto-7.0.25.tar.gz
  7.0.24    https://www.x.org/archive/individual/proto/xproto-7.0.24.tar.gz
  7.0.23    https://www.x.org/archive/individual/proto/xproto-7.0.23.tar.gz
  ...
  2.2.2     https://www.x.org/archive/individual/proto/dmxproto-2.2.2.tar.gz

==> How many would you like to checksum? (default is 1, q to abort) q
==> Error: Aborted.
```
Confirmed that this will be fixed by #2972.
